doc/content/integrations/cloud-integrations/thingsboard/thingsboard-setup/_index.md

```diff
@@ -0,0 +1,97 @@
+---
+title: "ThingsBoard Setup"
+description: ""
+weight: 1
+---
+
+After configuring the uplink payload formatter on {{% tts %}}, follow the steps in this section to prepare the setup on ThingsBoard. 
+
+<!--more-->
+
+Log in to your ThingsBoard account. 
+
+Before creating an integration, you need to define uplink and downlink data converters. To do so, first navigate to **Data converters** on the left hand menu.
+
+Click the **+** icon in the upper right and then select **Create new converter** to create a new uplink converter.
+
+Give a **Name** to your uplink converter, select **Uplink** as a **Type** and paste the following code in the decoder function field:
+
+```js
+var data = decodeToJson(payload);
+var deviceName = data.end_device_ids.device_id;
+var deviceType = data.end_device_ids.application_ids.application_id;
+
+var result = {
+    deviceName: deviceName,
+    deviceType: deviceType,
+    telemetry: {
+         temperature: data.uplink_message.decoded_payload.temperature
+    }
+};
+
+function decodeToString(payload) {
+    return String.fromCharCode.apply(String, payload);
+}
+
+function decodeToJson(payload) {
+    var str = decodeToString(payload);
+    var data = JSON.parse(str);
+    return data;
+}
+
+return result;
+```
+
+>**Note:** You can test the decoder by clicking the **Test decoder function** and submitting an uplink message from {{% tts %}} manually.
+
+{{< figure src="uplink-converter.png" alt="Creating the uplink converter" >}}
+
+Finish by selecting **Add**.
+
+Follow the same procedure to create a converter with **Downlink** as a **Type** and a following encoder function:
+
+```js
+var data = {
+        downlinks: [{
+            f_port: 2,
+            confirmed: false,
+            frm_payload: btoa(msg.version),
+            priority: "NORMAL"
+        }]
+    };
+
+var result = {
+    contentType: "JSON",
+    data: JSON.stringify(data),
+    metadata: {
+        devId: 'device_id'     //enter your device's ID here
+    }
+};
+return result;
+```
+
+>**Note:** Replace the `device_id` in the downlink converter function with the **End device ID** value from {{% tts %}}.
+
+{{< figure src="downlink-converter.png" alt="Creating a downlink converter" >}}
+
+Next, navigate to the **Integrations** section on the left hand menu. Add a new integration by clicking the **+** button in the upper right corner.
+
+Give a name to your integration by filling in the **Name** field. For **Type**, choose **TheThingsIndustries**.
+
+>**Note:** Make sure the **Enabled** and **Allow create devices or assets** checkboxes are left ticked.
+
+For **Uplink data converter** and **Downlink data converter** choose the uplink and downlink converters you created in the previous step.
+
+Depending on your {{% tts %}} deployment, choose **Host type**.
+
+If you want to use TLS, enter `8883` for **Port**. Otherwise, enter `1883`.
+
+Use the credentials of your {{% tts %}} MQTT Server as **Username** and **Password**.
```

Comment:
Is this clear? Shouldn't this be application ID and API key, that they can find in the Console view?

Reply:
Well there's this in the file before:

>  Also, note your credentials available under **MQTT** submenu of the **Integrations** menu on the left, because you will need them for further steps.

but yeah I can rephrase a bit. Although, I think it's more clear to tell those are the credentials for connecting to the MQTT server, than mentioning the appID and API key.

Reply:
Looks okay to me, there's instructions in the section before for obtaining your MQTT credentials.